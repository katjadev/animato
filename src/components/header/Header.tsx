import { FC } from 'react'
import { useTranslations } from 'next-intl'
import Logo from '@animato/components/logo/Logo'
import Icon from '@animato/components/icon/Icon'
import IconButton from '@animato/components/icon-button/IconButton'
import styles from './Header.module.css'

interface HeaderProps {
  title: string;
  className?: string;
}

const Header: FC<HeaderProps> = ({ title, className }) => {
  const t = useTranslations('project-header')
  
  return (
    <header className={`${styles.header} ${className}`}>
      <div className={styles.left}>
        <Logo variant='compact' />
        <IconButton ariaLabel={t('undo')}><Icon icon='undo' /></IconButton>
        <IconButton ariaLabel={t('redo')}><Icon icon='redo' /></IconButton>
      </div>
      <h1 className={styles.heading}>{title}</h1>
      <IconButton 
        ariaLabel={t('profile')}
        onClick={() => {}}
      >
        <Icon icon='profile-circle' />
      </IconButton>
    </header>
  )
}

export default Header