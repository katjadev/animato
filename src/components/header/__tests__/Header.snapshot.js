import { render } from '@testing-library/react'
import Header from '../Header'

describe('Header', () => {
  it('renders correctly', () => {
    const { container } = render(
      <Header
        title='Test Project'
        isAuthenticated={true}
        translations={{
          profile: 'Profile',
          undo: 'Undo',
          redo: 'Redo',
        }}
      />
    )
    expect(container).toMatchSnapshot()
  })
})
