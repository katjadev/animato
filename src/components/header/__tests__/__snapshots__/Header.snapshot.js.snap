// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div>
  <header
    class="header "
  >
    <div
      class="left"
    >
      <button
        aria-haspopup="true"
        class="button"
      >
        <img
          alt="Animato Logo"
          class="image"
          data-nimg="1"
          decoding="async"
          fetchpriority="high"
          height="32"
          src="/logo.svg"
          style="color: transparent;"
          width="32"
        />
        <i
          class="iconoir-nav-arrow-down "
        />
      </button>
      <button
        aria-disabled="true"
        aria-label="Undo"
        class="iconButton  medium"
      >
        <i
          class="iconoir-undo "
        />
      </button>
      <button
        aria-disabled="true"
        aria-label="Redo"
        class="iconButton  medium"
      >
        <i
          class="iconoir-redo "
        />
      </button>
    </div>
    <h1
      class="heading"
    >
      <span
        class="text"
        tabindex="0"
      />
    </h1>
    <div
      class="right"
    >
      <div
        class="autosave"
      >
        <i
          class="iconoir-cloud-check "
        />
      </div>
      <button
        aria-label="Profile"
        class="iconButton  medium"
      >
        <i
          class="iconoir-profile-circle "
        />
      </button>
    </div>
  </header>
</div>
`;
